### ISSUE TITLE
[UX] Breadcrumb design issues

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#2260]
From @klonos in #2246 

> 1. Please don't get me wrong. The breadcrumb looks OK when there are multiple crumbs, but here's how it currently looks when there is only one crumb and no secondary tabs (that's my 1.5.0 local with the primary tabs visible since I'm logged in):

> ![backdrop-basis-breadcrumb-only_one_crumb](https://cloud.githubusercontent.com/assets/2423362/18978463/cbe13ebc-8706-11e6-8519-888b7f3544df.png)

> (notice how the eye needs to move up and down to follow the trail)

> ...and here's how it looks in the PR sandbox (not logged in, so no primary tabs):

> ![backdrop-basis-breadcrumb-only_one_crumb-pr1592-not_logged_in](https://cloud.githubusercontent.com/assets/2423362/18977497/23a79e6a-86fe-11e6-9be3-0ab5dee2b130.png)

> The PR is better (less visual emphasis), but still not that great IMO. It's just a horizontal bar across the screen that draws unnecessary attention. Which brings me to...

> 2. In my head, I want the combination of breadcrumb > page title > primary tabs > secondary tabs to be a visual trail of where I am at. The way we have it now, it is page title > primary tabs > secondary tabs > breadcrumb. I don't think that this works that well. The breadcrumb should not come between the title/tabs section and content if you ask me.

> ...we are doing it so much better in the admin layout:

> ![backdrop-admin_layout-breadcrumb_trail_flow](https://cloud.githubusercontent.com/assets/2423362/18978515/1fb8fa20-8707-11e6-8323-64318d45f965.png)

> You can see that clearly if you switch using Basis as the admin theme:

> ![backdrop-admin_layout-breadcrumb_trail_flow-basis_as_admin_theme](https://cloud.githubusercontent.com/assets/2423362/18978558/5c09736a-8707-11e6-87c7-5b3f157590f5.png)

> I'm not saying that the breadcrumb belongs in the header, but it needs to be coming **before** the page title. Unfortunately, the page title pseudo-region (actually a pseudo-block) cannot be moved around (#933).

From @olafgrabienski in #2246:
> In my opinion, the situation when there is only one crumb, is even misleading. Unless you have a more nested page structure on a website, the breadcrumb looks like there is just another "Home" link on the page. Adding the current page as an unlinked crumb would improve the situation a lot.

> This (as the position of the breadcrumb) seems to be theme agnostic though. Structurally, you get the same result in Bartik at least.

**Comment by wesruv** [comment]
Agree sooooo much about breadcrumb never being "Home" and nothing else. I'm thinking the current page title should always be appended as plain text, can't think of a scenario where that's not an improvement.

I'm not sure why the Primary/Secondary tabs wouldn't also be in the breadcrumb, from a user perspective. I know from a technical perspective breadcrumbs are based on menus and maybe this is hairy.

BUUUUUT

If we're editing the About page, I think this is a reasonable and informative breadcrumb:

[Home](#home) > [About](#about) > Edit

Or in the last screenshot from @klonos, I would expect:

[Home](#home) > [Administration](#admin) > [Structure](#structure) > [Content types](#ct) > [Page](#page) > [Manage Display](#md) > Teaser

I'd probably want to concatenate a few of those middle crumbs with JS, but I think it's helpful, especially  "Structure >" and after.

**Comment by wesruv** [comment]
Definitely see all of @klonos points about eye flow, that's not what I was focusing on with the theme and it didn't even occur to me. I think it would be a template override or a default config update? Maybe both?

**Comment by klonos** [comment]
> ...that's not what I was focusing on with the theme and it didn't even occur to me.

And I never mentioned anything because we were so close to release date for 1.5.0 and I'd hate to derail things with what could possibly end up being an endless discussion.

> I think it would be a template override or a default config update? Maybe both?

Course of action I believe will depend on whether we consider this a UX bug or UX feature/task. I personally see it as a bug/oversight.

**Comment by jenlampton** [comment]
> I'm thinking the current page title should always be appended as plain text, can't think of a scenario where that's not an improvement.

Yes please, I do this manually on 90% of my sites! (It can be done in basis as part of `theme_breadcrumb()`.) Totally possible to do this for 1.6. See https://github.com/backdrop/backdrop-issues/issues/2402

> I'm not sure why the Primary/Secondary tabs wouldn't also be in the breadcrumb, from a user perspective. I know from a technical perspective breadcrumbs are based on menus and maybe this is hairy.

Yes, this is hairy. Primary/Secondary tabs are in the menu too, so it *should* be technically possible. Though, this is going to be hard, and not likely possible for 1.6. Removing milestone.

I think the weirdest part of the current layout is that the breadcrumbs are below the title + tabs. This was the case in Bartik too (which is why it's like that on all the layouts) but wasn't quite such an issue because of the design. I think the best solution to this general problem will be https://github.com/backdrop/backdrop-issues/issues/1704
